code/computational_geometry/src/sphere_tetrahedron_intersection/sphere_tetrahedron_intersection.cpp

```diff
@@ -0,0 +1,1237 @@
+	{
+		angRBF = std::acos( LUV::Dot( posR - posB, posC - posB ) );
+		angBFR = std::asin( lenB * std::sin( angRBF ) / radius );
+		angFRB = pi - ( angRBF + angBFR );
+		
+		prlA = radius;
+		prlB = radius;
+		prlC = radius;
+		prlD = std::sqrt( radius * radius - lenA * lenA );
+		prlE = prlD;
+		prlF = lenB * std::sin( angFRB ) / sin( angBFR );
+
+		prpA = posR + prdA * prlA;
+		prpB = posR + prdB * prlB;
+		prpC = posR + prdC * prlC;
+		prpD = posA + prdD * prlD;
+		prpE = posA + prdE * prlE;
+		prpF = posB + prdF * prlF;
+        
+		vecA = prpA - posR;
+		vecB = prpB - posR;
+		vecC = prpC - posR;
+		vecD = prpD - posR;
+		vecE = prpE - posR;
+		vecF = prpF - posR;
+		
+		dirA = vecA / LUV::Length( vecA );
+		dirB = vecB / LUV::Length( vecB );
+		dirC = vecC / LUV::Length( vecC );
+		dirD = vecD / LUV::Length( vecD );
+		dirE = vecE / LUV::Length( vecE );
+		dirF = vecF / LUV::Length( vecF );
+		
+		angERA = std::acos( LUV::Dot( dirE, dirA ) );
+		Vec3 vecAF = prpF - posA;
+		Vec3 vecAE = prpE - posA;
+		angFAE = std::acos( LUV::Dot( vecAF, vecAE ) / ( LUV::Length( vecAF ) * LUV::Length( vecAE ) ) );
+	}
+	
+public:
+
+	ObservationTetrahedron( const Vec3& vR, const Vec3& vA, const Vec3& vB, const Vec3& vC )
+	{
+		posR = vR;
+		posA = vA;
+		posB = vB;
+		posC = vC;
+		
+		Vec3 AmR = vA - vR;
+		Vec3 BmR = vB - vR;
+		Vec3 CmR = vC - vR;
+		Vec3 BmA = vB - vA;
+		Vec3 CmA = vC - vA;
+		Vec3 CmB = vC - vB;
+		
+		lenA = LUV::Length( AmR );
+		lenB = LUV::Length( BmR );
+		lenC = LUV::Length( CmR );
+		
+		prdA = AmR / lenA;
+		prdB = BmR / lenB;
+		prdC = CmR / lenC;
+		prdD = BmA / LUV::Length( BmA );
+		prdE = CmA / LUV::Length( CmA );
+		prdF = CmB / LUV::Length( CmB );
+		
+		angBAC = std::acos( LUV::Dot( prdD, prdE ) );
+		solidAngleFull = SolidAngleWedge( prdA, prdB, prdC );
+	}
+	
+	double GetSolidAngle( double radius )
+	{
+		Recalculate( radius );
+		if( radius > lenC )
+		{
+			return 0;
+		}
+		else if( radius > lenB )
+		{
+			return SolidAngleWedge( dirA, dirC, dirF )
+				- SolidAngleCap( angERA ) * angFAE / ( 2.0 * pi );
+		}
+		else if( radius > lenA )
+		{
+			return solidAngleFull
+				- SolidAngleCap( angERA ) * angBAC / ( 2.0 * pi );
+		}
+		else
+		{
+			return solidAngleFull;
+		}
+	}
+	
+	double GetSurfaceArea( double radius )
+	{
+		return GetSolidAngle( radius ) * radius * radius;
+	}
+	
+	double GetVolume( double radius )
+	{
+		Recalculate( radius );
+		if( radius > lenC )
+		{
+			return VolumeTetrahedron( posR, posA, posB, posC );
+		}
+		else if( radius > lenB )
+		{
+			return ( VolumeWedge( dirA, dirC, dirF, radius ) - (
+				VolumeCap( radius, LUV::Length( prpA - posA ) )
+				- VolumeCone( prlE, lenA )
+				) * angFAE / ( 2.0 * pi ) )
+				+ VolumeTetrahedron( posR, posA, posB, prpF );
+		}
+		else if( radius > lenA )
+		{
+			return VolumeWedge( dirA, dirB, dirC, radius ) - (
+				VolumeCap( radius, LUV::Length( prpA - posA ) )
+				- VolumeCone( prlE, lenA )
+				) * angBAC / ( 2.0 * pi );
+		}
+		else
+		{
```

Comment:
Our style guide says that single line branches do not require braces.

Reply:
Yeah, even I personally agree with you, but still, it is what the style guide says.